bundle/src/files.rs

```diff
@@ -45,125 +151,144 @@ pub struct FileSet {
 }
 
 impl FileSet {
-    /// Scan a file set from a glob path.
-    /// And generates file set using file counter.
-    ///
-    pub fn scan_from_glob(
-        repo_root: &str,
-        glob_with_status: JunitReportFileWithStatus,
-        file_counter: &mut FileSetCounter,
-        team: Option<String>,
-        codeowners: &Option<CodeOwners>,
-        start: Option<SystemTime>,
-    ) -> anyhow::Result<FileSet> {
-        let JunitReportFileWithStatus {
-            junit_path: glob_path,
-            status: resolved_status,
-        } = glob_with_status;
-        let path_to_scan = if !std::path::Path::new(&glob_path).is_absolute() {
-            std::path::Path::new(repo_root)
-                .join(&glob_path)
-                .to_str()
-                .ok_or_else(|| anyhow::Error::msg("failed to convert path to string"))?
-                .to_string()
-        } else {
-            glob_path.clone()
-        };
+    pub fn new(
+        files: Vec<BundledFile>,
+        glob: String,
+        resolved_status: Option<JunitReportStatus>,
+    ) -> Self {
+        Self {
+            file_set_type: FileSetType::Junit,
+            files,
+            glob,
+            resolved_status,
+        }
+    }
+}
 
-        let mut files = Vec::new();
+#[derive(Debug, Serialize, Deserialize, Clone, Default, PartialEq, Eq)]
+#[cfg_attr(feature = "pyo3", gen_stub_pyclass_enum, pyclass(eq, eq_int))]
+#[cfg_attr(feature = "wasm", derive(Tsify))]
+pub enum FileSetType {
+    #[default]
+    Junit,
+}
 
-        glob::glob(&path_to_scan)?.try_for_each(|entry| {
-            let path = match entry {
-                Ok(path) => path,
-                Err(e) => {
-                    return Err(anyhow::anyhow!("Error scanning file set: {:?}", e));
-                }
-            };
+#[cfg(feature = "wasm")]
+// u128 will be supported in the next release after 0.2.95
+#[derive(Debug, Serialize, Deserialize, Clone, Default, PartialEq, Eq)]
+#[cfg_attr(feature = "pyo3", gen_stub_pyclass, pyclass(get_all))]
+#[cfg_attr(feature = "wasm", derive(Tsify))]
+pub struct BundledFile {
+    pub original_path: String,
+    /// Added in v0.5.33
+    pub original_path_rel: Option<String>,
+    pub path: String,
+    pub owners: Vec<String>,
+    pub team: Option<String>,
+}
```

Comment:
Pretty much everything below this is moved from `types.rs` because it's only relevant to the structs and logic in `files.rs`